pkg/controller/handlers/slackreceiver/slackreceiver.go

```diff
@@ -47,3 +71,146 @@ func CreateOAuthApp(req router.Request, _ router.Response) error {
+	socketmodeHandler.Handle(socketmode.EventTypeConnectionError, middlewareConnectionError)
+	socketmodeHandler.Handle(socketmode.EventTypeConnected, middlewareConnected)
+
+	// Handle a specific event from EventsAPI
+	socketmodeHandler.HandleEvents(slackevents.AppMention, h.middlewareAppMentionEvent)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	s.subscribed[slackReceiver.Name] = cancel
+
+	go func() {
+		if err := socketmodeHandler.RunEventLoopContext(ctx); err != nil {
+			fmt.Printf("error running event loop: %v", err)
+		}
+		delete(s.subscribed, slackReceiver.Name)
+		resp.RetryAfter(time.Second * 10)
+	}()
+
+	return nil
+}
+
+func (s *Handler) UnsubscribeFromSlackEvents(req router.Request, _ router.Response) error {
+	slackReceiver := req.Object.(*v1.SlackReceiver)
+	s.subscribed[slackReceiver.Name]()
+
+	delete(s.subscribed, slackReceiver.Name)
+	return nil
+}
+
+type eventHandler struct {
+	ctx     context.Context
+	storage storage.Client
+}
+
+func (h *eventHandler) middlewareAppMentionEvent(evt *socketmode.Event, slackClient *socketmode.Client) {
+	eventsAPIEvent, ok := evt.Data.(slackevents.EventsAPIEvent)
+	if !ok {
+		return
+	}
+
+	slackClient.Ack(*evt.Request)
+
+	ev, ok := eventsAPIEvent.InnerEvent.Data.(*slackevents.AppMentionEvent)
+	if !ok {
+		return
+	}
+
+	if ev.BotID != "" {
+		return
+	}
+
+	var slackTriggers v1.SlackTriggerList
+	if err := h.storage.List(h.ctx, &slackTriggers, client.MatchingFields{
+		"spec.appID":  eventsAPIEvent.APIAppID,
+		"spec.teamID": eventsAPIEvent.TeamID,
+	}); err != nil {
+		fmt.Printf("failed to list slack triggers: %v", err)
+		return
+	}
+
+	var errs []error
+	for _, trigger := range slackTriggers.Items {
+		var workflows v1.WorkflowList
+		if err := h.storage.List(h.ctx, &workflows, client.MatchingFields{
+			"spec.threadName": trigger.Spec.ThreadName,
+			"spec.slack":      "true",
+		}); err != nil {
+			fmt.Printf("failed to list workflows: %v", err)
+			return
+		}
+
+		var payload = &strings.Builder{}
+		if err := json.NewEncoder(payload).Encode(map[string]interface{}{
+			"type":  "slack",
+			"event": ev,
+		}); err != nil {
+			fmt.Printf("failed to encode payload: %v", err)
+			return
+		}
+
+		for _, workflow := range workflows.Items {
+			err := h.storage.Create(h.ctx, &v1.WorkflowExecution{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: system.WorkflowExecutionPrefix,
+					Namespace:    workflow.Namespace,
+				},
+				Spec: v1.WorkflowExecutionSpec{
+					Input:        payload.String(),
+					ThreadName:   workflow.Spec.ThreadName,
+					WorkflowName: workflow.Name,
+				},
+			})
+			if err != nil {
+				errs = append(errs, err)
+			}
+		}
+	}
+
+	if len(errs) > 0 {
+		fmt.Printf("failed to create workflow executions: %v", errors.Join(errs...))
+	}
+}
+
+func middlewareConnecting(_ *socketmode.Event, _ *socketmode.Client) {
+	log.Println("Connecting to Slack with Socket Mode...")
```

Comment:
Do we need these logs? They're probably useful for debugging.

If so, can we use the package logger in Obot instead of `log`?

Reply:
yeah, switched to use the log package. These log are useful for debugging.